spackbot/pr_add_labels.py

```diff
@@ -0,0 +1,168 @@
+    "R": {"package": [r"^r$", r"^r-"]},
+    #
+    # Package status
+    #
+    "new-package": {
+        "filename": r"^var/spack/repos/builtin/packages/[^/]+/package.py$",
+        "status": r"^added$",
+    },
+    "update-package": {"status": [r"^modified$", r"^renamed$"]},
+    #
+    # Variables
+    #
+    "maintainers": {"patch": r"[+-] +maintainers +="},
+    #
+    # Directives
+    #
+    "new-version": {"patch": r"\+ +version\("},
+    "conflicts": {"patch": r"\+ +conflicts\("},
+    "dependencies": {"patch": r"\+ +depends_on\("},
+    "extends": {"patch": r"\+ +extends\("},
+    "virtual-dependencies": {"patch": r"\+ +provides\("},
+    "patch": {"patch": r"\+ +patch\("},
+    "new-variant": {"patch": r"\+ +variant\("},
+    "resources": {"patch": r"\+ +resource\("},
+    #
+    # Functions
+    #
+    "external-packages": {"patch": r"[+-] +def determine_spec_details\("},
+    "libraries": {"patch": r"[+-] +def libs\("},
+    "headers": {"patch": r"[+-] +def headers\("},
+    "smoke-tests": {"patch": r"[+-] +def test\("},
+    #
+    # Core spack
+    #
+    "architecture": {
+        "filename": r"^lib/spack/spack/(architecture|operating_systems|platforms)"
+    },
+    "binary-packages": {"filename": r"^lib/spack/spack/binary_distribution"},
+    "build-environment": {"filename": r"^lib/spack/spack/build_environment"},
+    "build-systems": {"filename": r"^lib/spack/spack/build_systems"},
+    "new-command": {
+        "filename": r"^lib/spack/spack/cmd/[^/]+.py$",
+        "status": r"^added$",
+    },
+    "commands": {
+        "filename": r"^lib/spack/spack/cmd/[^/]+.py$",
+        "status": [r"^modified$", r"^modified$"],
+    },
+    "compilers": {"filename": r"^lib/spack/spack/compiler"},
+    "directives": {"filename": r"^lib/spack/spack/directives"},
+    "environments": {"filename": r"^lib/spack/spack/environment"},
+    "fetching": {"filename": r"^lib/spack/spack/(fetch|url|util/url|util/web)"},
+    "locking": {"filename": r"^lib/spack/(spack|llnl)/util/lock"},
+    "modules": {"filename": r"^lib/spack/spack/modules"},
+    "stage": {"filename": r"^lib/spack/spack/stage"},
+    "tests": {"filename": r"^lib/spack/spack/test"},
+    "utilities": {"filename": r"^lib/spack/spack/util"},
+    "versions": {"filename": r"^lib/spack/spack/version"},
+    #
+    # Documentation
+    #
+    "documentation": {"filename": r"^lib/spack/docs"},
+    #
+    # GitHub
+    #
+    "travis": {"filename": r"^\.travis"},
+    "actions": {"filename": r"^\.github/actions"},
+    "workflow": {"filename": r"^\.github/workflows"},
+    "git": {"filename": r"^\.gitignore"},
+    "flake8": {"filename": r"^\.flake8"},
+    "licenses": {"filename": r"^LICENSE"},
+    "gitlab": {"filename": r"^share/spack/gitlab"},
+    #
+    # Other
+    #
+    "defaults": {"filename": r"^etc/spack/defaults"},
+    "vendored-dependencies": {"filename": r"^lib/spack/external"},
+    "sbang": {"filename": r"sbang"},
```

Comment:
Not entirely sure if this is still needed now that sbang has its own separate repo, but won't hurt

Reply:
we still need this, I think, as we copy it into the spack repo.